.github/workflows/build.yml

```diff
@@ -106,14 +118,14 @@ jobs:
           settarget x86_64-freebsd9 make freebsd64
 
       - name: Upload FreeBSD native libraries
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v3
         with:
           name: freebsd
           path: src/main/c/resources/native/freebsd
 
   java:
     name: Java compilation
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
```

Comment:
I would suggest pinning versions, generally as it ensures that a previous build can be re-built, later in time when "latest" changes

Reply:
I was actually hoping to use evergreen runner versions as a canary to help _catch_ places where the build process accidentally relies on the default version of something in the runner image. IMO because GitHub frequently update tools within the runner images, just pinning the runner version alone doesn't guarantee much stability. I'm hoping that the other changes I've made make the runner version irrelevant to reproducibility:

* The Linux/Windows build step now runs inside an Ubuntu 18.04 container, and so we're using the fixed version of GCC 8 shipped for that environment.
* Same story for the FreeBSD build step; all of the meaningful work runs in the cross-compiler container.
* The macOS build now specifically targets Mac OS X 10.7 at minimum, so (in theory) it doesn't matter what version of Xcode or macOS SDK you're using. Apple clang still happily accepts `-target` values for long-EOL'd OSes (right down to 10.0), so I think we should be fairly able to use just about any Xcode tools/macOS SDK and get a functionally-identical (if admittedly not byte-identical) product.
* The Java build doesn't use any native tools at all. We configure a Java 8 JDK before we hop into Gradle, and that should be the end of the reliance on the surrounding, non-Java environment. And even if we didn't do that, the Gradle build definition specifically requires a Java 8 toolchain such that if Gradle wasn't launched by a Java 8 JDK, it will go off, download one, and use that to make the builds. (It's really neat to watch it do it, actually – launch the build from a Java 17 environment and it goes off and fetches a compiler like it would any other dependency.)

I hear where you're coming from, though. It's tedious when a PR gets held up because the build system has shifted underneath you. I can change the jobs to `ubuntu-22.04`/`macos-12` if you'd prefer.

Reply:
cool, that works, thanks.